step_impl/step_impl.py

```diff
@@ -1,53 +1,31 @@
-from getgauge.python import step, before_scenario, Messages
-from jira import JIRA
+from getgauge.python import step
+from step_impl.jira import Jira
 import os
 
 
 def jira_admin():
-    return jira(os.getenv("JIRA_ADMIN_USERNAME"), os.getenv("JIRA_ADMIN_PASSWORD"))
+    return Jira(os.getenv("JIRA_ADMIN_USERNAME"), os.getenv("JIRA_ADMIN_PASSWORD"))
 
 
 def internal_employee():
-    return jira(
+    return Jira(
         os.getenv("INTERNAL_EMPLOYEE_USERNAME"), os.getenv("INTERNAL_EMPLOYEE_PASSWORD")
     )
 
 
-def jira(username, password):
-    return JIRA(
-        auth=(username, password),
-        options={"server": os.getenv("JIRA_BASE_URL")},
-    )
-
-
-def is_group(group_name):
-    return len(jira_admin().group_members(group_name)) >= 0
-
-
-def is_internal_employee_in_internal_employees_group():
-    internal_employees = jira_admin().group_members("jira-internal-employee_users")
-    return bool(
-        [
-            emp
-            for emp in internal_employees.values()
-            if (emp["name"] == os.getenv("INTERNAL_EMPLOYEE_USERNAME"))
-        ]
-    )
-
-
 @step("An internal employee can see all projects")
```

Comment:
Because there are a lot of layers in the python module and in the Jira logic over which we don't have control and if they change the simple checking of the same number of projects may lead to false positives. It might be safer to check if they can see the same number projects and if their project id's match:
@step("An internal employee can see all projects")
def an_internal_employee_can_see_all_projects():
	internal_emp_proj = [_.raw['id'] for _ in Bot.internal_employee.projects()]
	admin_emp_proj = [_.raw['id'] for _ in Bot.jira_admin.projects()]
	assert len([_ for _ in admin_emp_proj if _ not in internal_emp_proj]) == 0